### PR TITLE
Add TLS support for metric beat http server

### PR DESCRIPTION
**Pull request by leopucci**
Solves https://github.com/elastic/beats/issues/11457

### CODE REVIEW

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -113,7 +113,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 - Add AWS s3_request metricset. {pull}10949[10949] {issue}10055[10055]
 - Add s3_daily_storage metricset. {pull}10940[10940] {issue}10055[10055]
 - Add `coredns` metricbeat module. {pull}10585[10585]
-
+- Add SSL support for metricbeat HTTP server {pull}11482[11482] {issue}11457[11457]
```
Comment (kaiyan-sheng) [inline CHANGELOG.next.asciidoc]: need a new line after this :-)

Comment (leopucci) [reply CHANGELOG.next.asciidoc]: Indeed it does! Thanks

File: metricbeat/helper/server/http/http.go [hunk 2]
```diff
@@ -130,6 +148,11 @@ func (h *HttpServer) handleFunc(writer http.ResponseWriter, req *http.Request) {
 
 	case "GET":
 		writer.WriteHeader(http.StatusOK)
-		writer.Write([]byte("HTTP Server accepts data via POST"))
+		if h.server.TLSConfig != nil && h.server.TLSConfig.Certificates != nil {
```
Comment (andrewkroh) [inline metricbeat/helper/server/http/http.go]: The request itself can tell you if it was received over TLS: `if req.TLS != nil`. Sorry to bother you about this line again, but the thought came to be late.

Comment (leopucci) [reply metricbeat/helper/server/http/http.go]: No problem. It looks better this way. Let me test it...

Comment (leopucci) [reply metricbeat/helper/server/http/http.go]: @andrewkroh Done, tested ok

**Comment by andrewkroh** [comment]
jenkins, test this

File: metricbeat/helper/server/http/http.go [hunk 3]
```diff
@@ -130,6 +148,11 @@ func (h *HttpServer) handleFunc(writer http.ResponseWriter, req *http.Request) {
 
 	case "GET":
 		writer.WriteHeader(http.StatusOK)
-		writer.Write([]byte("HTTP Server accepts data via POST"))
+		if req.TLS != nil {
+			writer.Write([]byte("HTTPS Server accepts data via POST"))
```
Comment (urso) [inline metricbeat/helper/server/http/http.go]: tip: `io.WriteString(writer, "...")`.

**Pull request merged**